src/config.rs

```diff
@@ -0,0 +1,153 @@
+pub mod storage;
+pub mod settings;
+
+use std::collections::HashMap;
+use serde_derive::Deserialize;
+use serde_json::Value;
+use wildmatch::WildMatch;
+use crate::config::settings::{Setting, SettingInfo};
+
+const SETTINGS_JSON: &str = include_str!("./config/settings.json");
+
+#[derive(Debug, Deserialize)]
+struct JsonEntry {
+    key: String,
+    #[serde(rename = "type")]
+    entry_type: String,
+    default: String,
+    description: String,
+}
+
+trait StorageAdapter {
```

Comment:
Any adapter that implements this can be used as a storage engine for the configuration store.